File: mobile_through_project_final/src/main/java/com/ssafy/cafe/model/service/ProductService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import java.util.Map;

import org.springframework.cache.annotation.Cacheable;

import com.ssafy.cafe.model.dto.Product;

public interface ProductService {
    /**
     * 모든 상품 정보를 반환한다.
     * @return
     */
    List<Product> getProductList();
    
    /**
     * 신규 상품 정보를 반환한다.
     * @return
     */
    List<Product> getNewProductList();
    
    /**
     * 인기 상품 정보를 반환한다.
     * @return
     */
    List<Product> getBestProductList();
    
    /**
     * 커피 상품 정보를 반환한다.
     * @return
     */
    List<Product> getCoffeeProductList();
    
    /**
     * 차 상품 정보를 반환한다.
     * @return
     */
    List<Product> getTeaProductList();
    
    /**
     * 쿠키 상품 정보를 반환한다.
     * @return
     */
    List<Product> getCookieProductList();
    
    /**
     * backend 관통 과정에서 추가됨
     * 상품의 정보, 판매량, 평점 정보를 함께 반환
     * @param productId
     * @return
     */
    List<Map<String, Object>> selectWithComment(Integer productId);
}

File: app/src/main/java/com/ssafy/cobaltcoffee/service/ProductService.kt
package com.ssafy.cobaltcoffee.service

import com.ssafy.cobaltcoffee.dto.Product
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ProductService {
    // 전체 상품의 목록을 반환한다.
    @GET("/rest/product/")
    fun getProductList(): Call<List<Product>>

    // 신규 상품의 목록을 반환한다.
    @GET("/rest/product/new")
    fun getNewProductList(): Call<List<Product>>

    // 인기 상품의 목록을 반환한다.
    @GET("/rest/product/best")
    fun getBestProductList(): Call<List<Product>>

    // 커피 상품의 목록을 반환한다.
    @GET("/rest/product/coffee")
    fun getCoffeeProductList() : Call<List<Product>>

    // 차 상품의 목록을 반환한다.
    @GET("/rest/product/tea")
    fun getTeaProductList(): Call<List<Product>>

    // 쿠키 상품의 목록을 반환한다.
    @GET("/rest/product/cookie")
    fun getCookieProductList(): Call<List<Product>>

    // 상품 정보 반환
    @GET("/rest/product/{productId}")
    fun getProductWithComments(@Path("productId") productId: String): Call<List<Map<String, Any>>>
}